File: PhaoHoaDemoPublic/GRAPHICS/Csharp/PhaoHoaDemo03/HatBan.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;

namespace PhaoHoaDemo03
{
    class HatBan : Hat
    {
        public HatBan(PointF toado, PointF vantoc, int thoigiansong, Brush cove):
            base(toado, vantoc, thoigiansong, cove)
        {

        }


        public override void Ve(Graphics g)
        {
            g.FillEllipse(Brushes.White, ToaDo.X - 4, ToaDo.Y - 4, 8, 8);
            g.FillEllipse(this.CoVe, ToaDo.X - 2, ToaDo.Y - 2, 4, 4);
        }
    }
}

File: PhaoHoaDemoPublic/GRAPHICS/Python/PhaoHoaDemo03/hat_ban.py
from hat import Hat
from global_helper import Global

class HatBan(Hat):
    def __init__(self, toado, vantoc, thoigiansong, mausac):
        self.ToaDo = toado  # tọa độ
        self.VanToc = vantoc  # vận tốc, đúng nghĩa thì dùng kiểu Vector sẽ tốt hơn
        self.ThoiGianSong = thoigiansong  # thời gian sống, thời gian sống <= 0 tức là đã chết

        self.tkshapeid = Global.CreateHatBan(self.ToaDo, mausac)